### PR TITLE
Track number ordering

### PR DESCRIPTION
**Pull request by Hyzual**
Related to phanan/koel#1089

Hi, this is my first pull request for koel !
The intent of this contribution is to order album song lists first by CD number, then by track number.

Help for testing:
- Given an album with tracks in 2 CDs
- When you browse that album, songs are ordered first by CD number, then by track number, so that the following order is followed:

1. CD 1. Track 1
2. CD 1. Track 2
3. CD 2. Track 1

- After browsing, when you sort again by track number (by clicking on the table header), the ordering still does not consider Disc number. This calls for a deeper refactoring and will need a separate contribution.

Note to reviewer:
I have tried to add unit-tests for everything I've touched. Please let me know if more need to be added. I am not used at all to the "no-semicolon" style so apologies if some slipped through, I intended to `eslint fix` them all. Also I am used to a coding style where variables are in snake_case and functions are in camelCase, let me know if that is a problem and I'll adapt.

I do intend on contributing to koel some more and hope to be able to in the future. I can't promise anything on the frequency of those contributions, as family life and day job do not leave much time...
I have experience working on PHP / VueJS codebases and I have a couple of ideas to make koel even better.

Looking forward to your feedback on this :)

### CODE REVIEW

File: js/tests/components/song/list.spec.js [hunk 1]
```diff
@@ -6,6 +6,8 @@ import { event } from '@/utils'
 import { songStore, queueStore } from '@/stores'
 import { playback } from '@/services'
 import { mock } from '@/tests/__helpers__'
+import { shallowMount } from "@vue/test-utils"
```
Comment (phanan) [inline js/tests/components/song/list.spec.js:9]: Please prefer single quotes over double quotes in JS.

File: js/tests/components/song/list.spec.js [hunk 2]
```diff
@@ -136,4 +138,128 @@ describe('components/song/list', () => {
...
+        propsData: {
+          items: songs,
+          type: "all-songs"
+        }
+      })
+
+      expect(orderBy).toHaveBeenCalledWith(expect.any(Array), "", 1)
+    })
+
+    it(`when it's an album's song list,
```
Comment (phanan) [inline js/tests/components/song/list.spec.js:171]: This doesn't read as a sentence to me. How about something like this?

```suggestion
    it('sorts songs by track and disc number in album song list')
```

File: js/tests/components/song/list.spec.js [hunk 3]
```diff
@@ -136,4 +138,128 @@ describe('components/song/list', () => {
...
+    })
+  })
+
+  describe(`sort()`, () => {
+    let orderBy
+    beforeEach(() => {
+      orderBy = jest.spyOn(filters, "orderBy")
+    })
+
+    it(`when sorting is disallowed, it will do nothing`, () => {
```
Comment (phanan) [inline js/tests/components/song/list.spec.js:194]: Same as above.

File: js/tests/components/song/list.spec.js [hunk 4]
```diff
@@ -136,4 +138,128 @@ describe('components/song/list', () => {
...
     wrapper.vm.filteredItems.forEach(item => expect(item.selected).toBe(true))
   })
+
+  describe(`mounted()`, () => {
+    let orderBy
+    beforeEach(() => {
+      orderBy = jest.spyOn(filters, "orderBy")
+    })
+
+    it(`when sorting is disallowed, it will not sort`, () => {
```
Comment (phanan) [inline js/tests/components/song/list.spec.js:148]: Same as other comments. Also please use present tense e.g. `it('does not sort when…')`

File: js/utils/filters.js [hunk 5]
```diff
@@ -1,36 +1,21 @@
-import { isObject, isNumber, get } from 'lodash'
+import { isObject, isNumber, get } from "lodash"
```
Comment (phanan) [inline js/utils/filters.js:1]: Similar to a previous comment, please revert this change.

**Comment by Hyzual** [comment]
Thanks for the review 😊, I'll adjust when I get the chance.

File: js/tests/components/song/list.spec.js (hunk 1 above)
Comment (Hyzual) [reply js/tests/components/song/list.spec.js]: Done. I suggest adding the [quotes](https://eslint.org/docs/rules/quotes) eslint rule. That way, it can be fixed automatically.
